node-admin/src/test/java/com/yahoo/vespa/hosted/node/admin/task/WriteFileTaskTest.java

```diff
@@ -0,0 +1,73 @@
+// Copyright 2018 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.vespa.hosted.node.admin.task;
+
+import org.glassfish.jersey.internal.util.Producer;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class WriteFileTaskTest extends TaskTestBase {
+    private final String content = "line1\nline2\n";
+
+    @Test
+    public void testWrite() {
+        Path path = Paths.get("foo");
+
+        @SuppressWarnings("unchecked")
+        Producer<String> contentProducer = (Producer<String>) mock(Producer.class);
```

Comment:
Why not just `Producer<String> contentProducer = () -> content;` ?

Reply:
I wanted to verify it was being called once: Added